test/deployframework/helpers.go

```diff
@@ -289,3 +284,291 @@ func cleanupLocalCmds(logger logrus.FieldLogger, commands ...exec.Cmd) error {
+	err := wait.Poll(1*time.Second, 15*time.Second, func() (done bool, err error) {
+		_, err = client.CoreV1().ConfigMaps(namespace).Get(name, meta.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to wait for the %s configmap to be created in the %s namespace: %v", name, namespace, err)
+	}
+	logger.Infof("The %s ConfigMap has been created in the %s namespace", name, namespace)
+
+	return nil
+}
+
+// UpdateExistingSubscription is a helper function responsible for upgrading an existing metering-ocp Subscription
+// to use the newest payload and verify that the Subscription object is reporting a successful upgrade status.
+func UpdateExistingSubscription(logger logrus.FieldLogger, client olmclientv1alpha1.OperatorsV1alpha1Interface, name, upgradeChannel, namespace string) error {
+	sub, err := client.Subscriptions(namespace).Get(name, meta.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return fmt.Errorf("the %s subscription does not exist", name)
+	}
+	if err != nil {
+		return err
+	}
+
+	// update the Subscription to use the most recent channel listed in the package.yaml
+	// and change the Subscription source type to use the contents of a CatalogSource.
+	sub.Spec.CatalogSource = name
+	sub.Spec.CatalogSourceNamespace = namespace
+	sub.Spec.Channel = upgradeChannel
+	_, err = client.Subscriptions(namespace).Update(sub)
+	if err != nil {
+		return err
+	}
+	logger.Infof("Updated the %s Subscription to use the %s channel", name, upgradeChannel)
+
+	// after updating the metering-ocp Subscription to use a newer channel,
+	// wait until this object is reporting a successful upgrade state before
+	// transferring control back to the function call site.
+	err = wait.Poll(3*time.Second, 1*time.Minute, func() (done bool, err error) {
+		sub, err := client.Subscriptions(namespace).Get(name, meta.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+
+		logger.Infof("Waiting for the %s Subscription to finish upgrading", name)
+		if !strings.Contains(sub.Status.CurrentCSV, upgradeChannel) {
+			logger.Infof("Subscription status does not report metering-operator-v%s as the currentCSV", upgradeChannel)
+			return false, nil
+		}
+		if sub.Status.State != olmv1alpha1.SubscriptionStateAtLatest {
+			logger.Infof("Subscription status has not reported AtLatestKnown yet")
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to wait for the %s subscription to finish updating in the %s namespace: %v", name, namespace, err)
+	}
+	return nil
+}
+
+// WaitForMeteringOperatorDeployment is a helper function that will poll for the @name
+// deployment to report a single integer for both of the UpdatedReplicas and Replicas
```

Comment:
actually it's checking for an integer typed value of 1 for both. The comment makes it seem like something else is going on.